### PR TITLE
Fix SHAMapStore test timing consistency

### PR DESCRIPTION
**Pull request by ximinez**
Reviewers: @miguelportilla @vinniefalco

### CODE REVIEW

File: src/ripple/app/misc/SHAMapStoreImp.cpp [hunk 1]
```diff
@@ -251,10 +251,24 @@ SHAMapStoreImp::onLedgerClosed(
     {
         std::lock_guard <std::mutex> lock (mutex_);
         newLedger_ = ledger;
+        working_ = true;
     }
     cond_.notify_one();
 }
 
+void
+SHAMapStoreImp::rendezvous() const
```
Comment (vinniefalco) [inline src/ripple/app/misc/SHAMapStoreImp.cpp]: Wow that didn't take long, now everyone is copying my tricks

Comment (ximinez) [inline src/ripple/app/misc/SHAMapStoreImp.cpp]: "Good programmers steal." :)

Comment (vinniefalco) [inline src/ripple/app/misc/SHAMapStoreImp.cpp]: I guess I can't complain since I'm against "intellectual property."

**Comment by vinniefalco** [comment]
Is the code coverage change real?

**Comment by ximinez** [comment]
Yep. Almost none of the `SHAMapStore` code is covered by other tests.

**Comment by nbougalis** [comment]
It looks like it is - the unit test is the only thing that exercises this code, and it was turned to a manual test before, so the code wasn't getting touched.

**Comment by vinniefalco** [comment]
Oh right, because the test was changed to manual. Sweet.

Anyway I am going to give this a thumbs up but I have concerns about the brittle code path between the call from `env::close` to `SHAMapStoreImp::onLedgerAccept`. Its not clear to someone who might work in a function in that call chain (`LedgerMaster::checkAccept` for example) that the operation must remain synchronous or else unit tests have a remote chance of failing.
:+1:

**Comment by ximinez** [comment]
Random data point: I was able to reproduce the original failure on Linux within an hour or so by running the tests in a tight loop. I (unintentionally) left the updated test running for over 41 hours in the same loop and didn't repro.

**Comment by ximinez** [comment]
Rebased against 0.32.0-b1 just to ensure there are no conflicts.

**Comment by miguelportilla** [comment]
Tested this unit test in a loop in OS X, Windows and Ubuntu and was unable to repro the failure. Changes LGTM :+1:

**Pull request merged**